EtherealEngine/etherealengine
# 9402 Batched Compression

## Summary
_( Charlie and Jeremy's first big feature!✨ )_

- The model transform properties panel now has a batch compression checkbox; when checked, compressed GLBs with texture size 2048 (`LOD_0`), 1024 (`LOD_1`) and 512 (`LOD_2`) are generated alongside the original file. (The selected model and the original GLTF/GLB are unchanged.)
- All this functionality is also now available when right-clicking a model in the file browser and choosing "Compress".

## References
closes #9402
closes #9418

## QA Steps

1. Load the Studio and open the `default project`.
2. In the Scene panel, choose the default scene. Wait for the scene to finish loading.
3. In the Files panel, navigate into the `assets` folder, find the file `apartment.glb`, and drag it into the scene or into the Hierarchy panel. Wait for the model to load and appear.
4. Select the model.
5. In the Properties panel, scroll to the `Model Transform Properties` section and expand it.
6. **Confirm** that the `Clientside Transform` checkbox is now checked by default.
7. **Confirm** that there is a new unchecked checkbox, `Batch Compress`.
8. Check `Batch Compress` and click `Optimize`.
9. Refresh the Files panel.
10. **Confirm** that `apartment-transformed-LOD_0.glb`, `apartment-transformed-LOD_1.glb` and `apartment-transformed-LOD_2.glb` appear in the panel.
11. In the Files panel, delete `apartment-transformed-LOD_0.glb`, `apartment-transformed-LOD_1.glb` and `apartment-transformed-LOD_2.glb`. _Do not_ delete `apartment.glb`.
12. In the Files panel, right-click `apartment.glb` and click `Compress`.
13. **Confirm** that a popup appears with the name `apartment`, and that the UI contains an element named `Model Format`.
14. Click `COMPRESS`. Wait for the popup to close.
15. **Confirm** that `apartment-transformed-LOD_0.glb`, `apartment-transformed-LOD_1.glb` and `apartment-transformed-LOD_2.glb` appear again in the panel. Delete them if you like.
